app/src/org/commcare/activities/FormAndDataSyncer.java

```diff
@@ -203,13 +204,13 @@ protected void deliverUpdate(CommCareHomeActivity receiver, Integer... update) {
             }
 
             @Override
-            protected void deliverError(CommCareHomeActivity receiver,
+            protected void deliverError(ConnectorWithMessaging receiver,
                                         Exception e) {
-                receiver.displayMessage(Localization.get("sync.fail.unknown"), true);
+                receiver.displayBadMessage(Localization.get("sync.fail.unknown"));
             }
         };
 
-        mDataPullTask.connect(activity);
+        mDataPullTask.connect((CommCareTaskConnector<ConnectorWithMessaging>)activity);
```

Comment:
Only line that jumps out at me /AS is complaining about. Possibly a comment explaining why this is safe?